File: Java/Fila/Queue.java
package Fila;

public class Queue<T> {

    Node<T> head;
    Node<T> tail;
    int length;

    public Queue() {
        this.head = null;
        this.tail = null;
        length = 0;
    }

    public boolean isEmpty() {
        if (length == 0) {
            return true;
        }

        return false;
    }

    public void enqueue(T value) {
        Node<T> node = new Node<T>(value);

        if (isEmpty()) {
            this.head = node;
            this.tail = node;
            length++;
            return;
        }

        this.tail.next = node;
        node.next = null;
        this.tail = node;
        length++;
    }

    public T dequeue() {

        if (isEmpty()) {
            return null;
        }

        T data;
        data = this.head.data;

        if (length == 1) {
            this.tail = null;
        }

        this.head = this.head.next;
        length--;
        return data;

    }

    public T peek() {
        if (isEmpty()) {
            return null;
        }

        T data;
        data = this.head.data;
        return data;
    }

}

File: Python/Fila/Queue.py
from typing import Generic, Optional, TypeVar
from Node import Node
T = TypeVar("T")

class Queue(Generic[T]):

    def __init__(self):
        self.head:Optional['Node[T]'] = None
        self.tail:Optional['Node[T]'] = None
        self.length = 0
    
    def isEmpty(self) -> bool:
        return self.head is None
    

    def enqueue(self, data : T):
        node = Node(data)

        if(self.isEmpty()):
            self.head = node
            self.tail = node
            self.length += 1
            return
        
        self.tail.next = node
        self.tail = node
        self.length += 1
        return
    
    def dequeue(self) -> T:
        if(self.isEmpty()):
            print("This queue is empty")
            return None
        nodeData = self.head.data
        if(self.length == 1):
            self.head = None
            self.tail = None
            self.length -= 1
            return nodeData
        
        self.head = self.head.next
        self.length -= 1
        return nodeData
    
    def peek(self) -> T:
        if(self.isEmpty()):
            print("This queue is empty")
            return None

        nodeData = self.head.data
        return nodeData